### ISSUE TITLE
`revng translate` fails with `error: PHDR segment not covered by LOAD segment`

### ISSUE THREAD

**Issue by delvershi** [revng/revng#26]
Hello,I have use a dynamic binary and try to translate it.But i get a coredump when run the translated binary:
```
1.  x86_64-gentoo-linux-musl-gcc hello.c -o hello.x86-64
2.  revng translate hello.x86-64
3.  ./hello.x86-64.translated
     ./hello.x86-64.translated: ./hello.x86-64.translated: no version information available (required by ./hello.x86-64.translated)
Segmentation fault (core dumped)
```
Dose the rebuild revng support the dynamic binary?
If Not,do you plan to support dynamic binary next?

**Comment by aleclearmind** [comment]
Please:

1. attach the binary you're using
2. report GCC and linker version (`gcc --version` and `ld --version`)
3. report distro you're using

Can you translate `/bin/bash`?

We might have an out-of-tree fix for this already, let me investigate this further.

Thanks.

**Comment by delvershi** [comment]
Hello,
1. the binary i generated:
[hello.zip](https://github.com/revng/revng/files/5671458/hello.zip)
1. my gcc version is ```gcc (GCC) 10.1.0```
    my linker version is ```GNU gold (GNU Binutils 2.35) 1.16```
2. orchestra : ```commit 306a5be2ef71f031a5e972e30c17afbcbabaa4b7 (HEAD -> master, origin/master, origin/HEAD)```
revng : ```commit b420128e9fdcbd33b1c38de54f03293266a90196 (HEAD -> develop, origin/develop, origin/HEAD, orchestra-temporary)```
3. i can't translate /bin/bash correctly
```
cp /bin/bash .
revng translate ./bash

/home/kkk/newRevng/orchestra/root/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/gcc-bin/10.1.0/../../../../lib/gcc/x86_64-pc-linux-gnu/10.1.0/../../../../x86_64-pc-linux-gnu/bin/ld.bfd: cannot find -ltinfo
collect2: error: ld returned 1 exit status
The following command exited with 1:
/home/kkk/newRevng/orchestra/root/x86_64-pc-linux-gnu/x86_64-pc-linux-gnu/gcc-bin/10.1.0/c++ ./bash.translated.ll.linked.ll.o -lz -lm -lrt -lpthread -L ./ -o ./bash.translated -fno-pie -no-pie -Wl,-z,max-page-size=4096 -Wl,--section-start=.o_rx_0x0=0x0 -Wl,--section-start=.o_rw_0x303d90=0x303d90 -fuse-ld=bfd -Wl,--section-start=.elfheaderhelper=0x303d8f -Wl,-Ttext-segment=0x31a000 -Wl,--no-as-needed -ltinfo -ldl -lc -Wl,--as-needed
```

**Comment by mmkp** [comment]
Hello,

I'm having a similar problem. 
When I try to translate the hello world example or the /bin/bash executable, I get the following error:
`
 /usr/bin/ld.bfd: bash.translated: error: PHDR segment not covered by LOAD segment
`
Could you please help me solve this?

Thank you.

**Comment by aleclearmind** [comment]
OK, sorry for the wait. This should be fixed soon.

**Comment by aleclearmind** [comment]
OK, the problem should be fixed in 09f4477. Basically the issue was that dynamic libraries (and PIE executables) where load at address 0, which is not OK.
If you want to try out bash please update QEMU too (see commit e196d27 there), but I still get failures since bash tries to install its own signal handler, which messes up with our mechanism, but that probably deserves another issue.

Closing this now, feel free to reopen if you're still experiencing this problem.
